mm2src/lp_ordermatch/order_requests_tracker.rs

```diff
@@ -0,0 +1,88 @@
+use std::{collections::hash_map::{HashMap, RawEntryMut},
+          num::NonZeroUsize,
+          time::Duration};
+use wasm_timer::Instant;
+
+const ONE_SECOND: Duration = Duration::from_secs(1);
+
+/// Stores the timestamps of order requests sent to specific peer
+pub struct OrderRequestsTracker {
+    requested_at: HashMap<String, Vec<Instant>>,
```

Comment:
Just an idea:
We can optimize the storage and operations upon it by changing `requested_at` type to `HashMap<String, (Instant, u8)>`, where the value is pair [timestamp, count_of_requests]
Then limit_reached() will check `if timestamp == now() && count_of_requests < 5`
Then peer_requested() will `increment the count_of_requests` or `set the timestamp to now() and count_of_requests to 1`

Reply:
Thanks for your review and comment!

This idea definitely worth considering, but we should note that it changes the actual logic of limiter. Imagine the time line with points on it meaning timestamp of requests. The current approach is actually a sliding 1 second window that always contains not more than 5 points. The suggested approach will divide the timeline into 1 second segments which will allow the 2x per second requests burst which might be unwanted, e.g.:

![Demo](https://user-images.githubusercontent.com/15745003/94107499-ccd7c200-fe67-11ea-9954-b1f8898c6b53.png)

On the picture you can see that we split the line to 1 second segments first and in each segment we have only 3 requests, but there's 1 second interval that has 6 points which is over allowed limit.

But yeah, we have to pay with O(n) memory usage overhead to have such precision.